internal/search/zoekt/indexed_search.go

```
@@ -434,7 +446,16 @@ func zoektGlobalQuery(q zoektquery.Q, repoOptions search.RepoOptions, userPrivat
 		apply(zoektquery.RcOnlyForks, repoOptions.OnlyForks)
 		apply(zoektquery.RcNoForks, repoOptions.NoForks)
 
-		qs = append(qs, zoektquery.NewAnd(&zoektquery.Branch{Pattern: "HEAD", Exact: true}, rc))
+		children := []zoektquery.Q{&zoektquery.Branch{Pattern: "HEAD", Exact: true}, rc}
+		for _, pat := range repoOptions.MinusRepoFilters {
+			re, err := regexp.Compile(`(?i)` + pat)
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid regex for -repo filter %q", pat)
+			}
```

side comment: this is the sort of thing where, if we guarantee `pat` is valid already (we do) and we perform an operation on pat that is always valid (I believe prepending `(?i)` is always valid) then we can design an interface around such guarantees and this function would not have to error, and none of the callers would either. Like, I believe we could make this a `MustCompile`, but that's a bit "raw" and not really indicative of the guarantees we have here.

A nicer abstraction would be to allow certain safe operations via functions/methods on this type that don't entertain the possibility of it erroring, and so we can remove the checks/propagation. Something for me to think about when we're closer to concentrating this logic together.

---

Yeah I see we just ignore the error in other parts of the code. TBH it is easy enough to return an error here + it is minimal error fluff + feels safer to not couple in general. I agree though if we had the correct interface (probably just `*regexp.Regexp` or `*syntax.Regexp`) that would be much better.